### PR TITLE
added output_schema to WITH clause

### PR DESCRIPTION
**Pull request by TobiasHauck**
### Description
Fixes #2619
 
Currently connect schemas in the schema registry are inferred by KSQL which means all field names become uppercase (except you explicitly write ```SELECT id as `id`, ...```), all fields become optional and Maps become arrays of MapEntries which is a result of all fields becoming optional. This behaviour is intended.

But sometimes it's undesirable. Let's say you want to filter a few properties of an existing topic containing avro data into a new topic, you expect that the new topic has the same field definitions than the filtered topic. But that's not the case. So we need more control over the resulting data schema.

This is the reason why I introduced a new WITH property called ```output_schema``` that can be used to tell the query which exact schema to use for the resulting topic and its data.

### Testing done 
- [x] Added missing unit tests
- [x] Checked that all tests & checkstyle still running properly
- [x] Built & ran the ksql server & cli locally as well as in a Docker container and created a few streams and tables manually by cli and/or rest client (Postman)

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: config/ksql-server.properties [hunk 1]
```diff
@@ -49,4 +49,4 @@ ksql.logging.processing.stream.auto.create=true
 bootstrap.servers=localhost:9092
 
 # Uncomment and complete the following to enable KSQL's integration to the Confluent Schema Registry:
-#ksql.schema.registry.url=?
+#ksql.schema.registry.url=http://localhost:8081
```
Comment (donjuandem) [inline config/ksql-server.properties]: dat muss denke ich so bleiben, wies war

File: docs/developer-guide/syntax-reference.rst [hunk 2]
```diff
@@ -444,6 +444,9 @@ The WITH clause for the result supports the following properties:
 |                         | ``JSON``, ``DELIMITED`` (comma-separated value), and ``AVRO``. If this property is not               |
 |                         | set, then the format of the input stream/table is used.                                              |
 +-------------------------+------------------------------------------------------------------------------------------------------+
+| OUTPUT_SCHEMA           | The name of the topic you want to copy your avro schema from. If this property is not set, then the  |
```
Comment (donjuandem) [inline docs/developer-guide/syntax-reference.rst]: Mit Topic hat das hier nix zu tun: DIe Topics werden ja durch die queries ers erstellt. Es is ja ein avro schema in der registry, dass du referenziers

File: ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlAvroTopicSerDe.java [hunk 3]
```diff
@@ -72,12 +79,14 @@ private static AvroConverter getAvroConverter(
       final KsqlConfig ksqlConfig,
       final Supplier<SchemaRegistryClient> schemaRegistryClientFactory,
       final String loggerNamePrefix,
-      final ProcessingLogContext processingLogContext
-  ) {
+      final ProcessingLogContext processingLogContext) {
+    final Schema finalSchema = (new OutputSchemaFactory(schemaRegistryClientFactory.get()))
```
Comment (donjuandem) [inline ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlAvroTopicSerDe.java]: nenn hier das, was schema is `maybeSchema` und finalSchema einfach `schema`. Dat würde in meinen augen mehr sinn machen

File: ksql-serde/src/main/java/io/confluent/ksql/serde/avro/OutputSchemaFactory.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Optional;
+import org.apache.avro.Schema.Parser;
+import org.apache.kafka.connect.data.Schema;
+
+public class OutputSchemaFactory {
+
+  private final SchemaRegistryClient srClient;
+
+  public OutputSchemaFactory(final SchemaRegistryClient srClient) {
+    this.srClient = srClient;
```
Comment (donjuandem) [inline ksql-serde/src/main/java/io/confluent/ksql/serde/avro/OutputSchemaFactory.java]: name is bs :D

Comment (TobiasHauck) [reply ksql-serde/src/main/java/io/confluent/ksql/serde/avro/OutputSchemaFactory.java]: die nennen das überall srClient. Deshalb bin ich hier dabei gelieben

Comment (donjuandem) [reply ksql-serde/src/main/java/io/confluent/ksql/serde/avro/OutputSchemaFactory.java]: Ah ok, dann is io :D

File: docs/developer-guide/syntax-reference.rst (hunk 2 above)
Comment (donjuandem) [reply docs/developer-guide/syntax-reference.rst]: „The name of the (existing) avro schema the topic created by this query should use. If this property is not set,   |
	|                         | the resulting schema will be inferred. Only supported in combination with  VALUE_FORMAT ``AVRO``       |“

File: docs/developer-guide/syntax-reference.rst [hunk 5]
```diff
@@ -444,6 +444,9 @@ The WITH clause for the result supports the following properties:
 |                         | ``JSON``, ``DELIMITED`` (comma-separated value), and ``AVRO``. If this property is not               |
 |                         | set, then the format of the input stream/table is used.                                              |
 +-------------------------+------------------------------------------------------------------------------------------------------+
+| OUTPUT_SCHEMA           | The name of the (existing) avro schema the topic created by this query should use. If this property  |
+|                         | is not set, the schema will be inferred. Only supported in combination with VALUE_FORMAT is ``AVRO`` |
```
Comment (donjuandem) [inline docs/developer-guide/syntax-reference.rst]: entweder `... with VALUE_FORMAT AVRO` oder `supported, if the VALUE_FORMAT is AVRO`

File: docs/developer-guide/syntax-reference.rst [hunk 6]
```diff
@@ -531,6 +534,9 @@ The WITH clause supports the following properties:
 |                         | ``JSON``, ``DELIMITED`` (comma-separated value), and ``AVRO``. If this property is not               |
 |                         | set, then the format of the input stream or table is used.                                           |
 +-------------------------+------------------------------------------------------------------------------------------------------+
+| OUTPUT_SCHEMA           | The name of the (existing) avro schema the topic created by this query should use. If this property  |
+|                         | is not set, the schema will be inferred. Only supported in combination with VALUE_FORMAT is ``AVRO`` |
```
Comment (donjuandem) [inline docs/developer-guide/syntax-reference.rst]: same here

**Pull request merged**